include/roboteam_ai/utilities/Dealer.h

```diff
@@ -48,12 +50,22 @@ class Dealer {
     };
 
    public:
+    struct FlagScore {
+        double score;
+        double weight;
+    };
+
+    struct RobotRoleScore {
+        double sumScore;
+        double sumWeights;
+    };
+
     struct DealerFlag {
         DealerFlagTitle title;
         DealerFlagPriority priority;
         explicit DealerFlag(DealerFlagTitle title, DealerFlagPriority priority);
     };
-    using FlagMap = std::map<std::string, std::vector<DealerFlag>>;
+    using FlagMap = std::map<std::string, std::pair<DealerFlagPriority, std::vector<DealerFlag>>>;
```

Comment:
std::unordered_map + wrap the pair in a struct like RobotRoleScore n stuff

```cpp
struct ... {
    DealerFlagPriority prio;
    std::vector<DealerFlag> flags;
};
```

Reply:
Will fix, didn't think of the idea that i could use a struct in a map